AsyncGuidance.md

```diff
@@ -144,6 +144,8 @@ Long running work in this context refers to a thread that's running for the life
 
 :bulb: **NOTE:`Task.Factory.StartNew` has an option `TaskCreationOptions.LongRunning` that under the covers creates a new thread and returns a Task that represents the execution. Using this properly requires several non-obvious parameters to be passed in to get the right behavior on all platforms.**
 
+:bulb: **NOTE: Don't use `TaskCreationOptions.LongRunning` with async code, it's will create a new thread, that will be destroyed after first `await`.**
```

Comment:
I wonder if this should be a rule by itself.

Reply:
Hi,

Would it be possible to get more information about this one? 
I used to use the following without problem but I may have misunderstood the statement above. 
This is used to run async operations in background that usually take 4-5 minutes. I've never seen the thread being destroyed after the first call to "await".

Thanks

```
Task.Factory.StartNew(async () => {
	try
	{
		await this();

		DoThat();

		await theSecond();
	}
	catch (Exception exc)
	{
		// do whatever you need with this
	}
}, cancellationToken, TaskCreationOptions.LongRunning, TaskScheduler.Default);
```

Reply:
It is destroyed after the first await. The underlying thread will terminate and the asynchronous operator will continue on another thread (likely a thread pool thread). Remember that asynchronous operations span multiple threads so just because you see your code continue, doesn’t mean it’s ujsfj the original thread it stated on.

Reply:
Ok that's what I thought... I got the statement wrong ;-)
What would you suggest using for such "long" running jobs (5-10 min)? 
new Thread(...)? Task.Run(...)?

Thanks David

Reply:
I assume phrase _thread is being destroyed_ confuses you. It is not to say that thread is destroyed and your code stops executing. When `Task.Factory.StartNew(...)` is called with `TaskCreationOptions.LongRunning` under the hood new dedicated thread is created to run the delegate. However, as soon as code execution reaches first `await` , it will continue executing on ThreadPool thread. And in ASP.NET Core [asynchronous continuations may run on any thread](https://blog.stephencleary.com/2017/03/aspnetcore-synchronization-context.html#beware-implicit-parallelism) so after `async` operation completes code won't continue executing on the initial thread. Thus, you created a thread (which is expensive) to only kick off `async` operation. Hence the guidance to not use  `Task.Factory.StartNew(...)` with `async` delegates.

Reply:
Thanks both, totally got it now.
I'll stick to Task.Run then since most, if not all, of my code is async "aware"